configure-memory-usage.md

```diff
@@ -58,19 +58,40 @@ In this configuration, when the memory usage of a tidb-server instance reaches 3
 
 ## Trigger the alarm of excessive memory usage
 
-In the default configuration, a tidb-server instance prints an alarm log and records associated status files when the machine memory usage reaches 80% of its total memory. You can set the memory usage ratio threshold by configuring `memory-usage-alarm-ratio`. For detailed alarm rules, refer to the description of [`memory-usage-alarm-ratio`](/tidb-configuration-file.md#memory-usage-alarm-ratio-new-in-v409).
+When the memory usage of a tidb-server instance exceeds its memory threshold (70% of its total memory by default) and any of the following conditions are met, TiDB records the associated status files and prints an alarm log.
 
-Note that after the alarm is triggered once, it will be triggered again only if the memory usage rate has been below the threshold for more than ten seconds and reaches the threshold again. In addition, to avoid storing excessive status files generated by alarms, currently, TiDB only retains the status files generated during the recent five alarms.
+- It is the first time the memory usage exceeds the memory threshold.
+- The memory usage exceeds the memory threshold and it has been more than 60s since the last alarm.
+- The memory usage exceeds the memory threshold and `(Current memory usage - Memory usage at the last alarm) / Maximum available memory > 10%`.
+
+You can control the memory threshold that triggers the alarm by modifying the memory usage ratio via the system variable [`tidb_memory_usage_alarm_ratio`](/system-variables.md#tidb_memory_usage_alarm_ratio).
+
+When the alarm of excessive memory usage is triggered, TiDB takes the following actions:
+
+- TiDB records the following information in the directory where the TiDB log files [`filename`](tidb-configuration-file.md#filename) are located.
+
+    - The information about the top 10 SQL statements with the highest memory usage and the top 10 SQL statements with the longest running time among all SQL statements currently being executed
+    - The goroutine stack information
+    - The usage status of heap memory
+
+- TiDB prints an alarm log containing the keyword `tidb-server has the risk of OOM` and the values of the following memory-related system variables.
+
+    - [`tidb_mem_oom_action`](#tidb_mem_oom_action-span-classversion-mark-v610-span)
+    - [`tidb_mem_quota_query`](#tidb_mem_quota_query)
+    - [`tidb_server_memory_limit`](#tidb_server_memory_limit)
+    - [`tidb_analyze_version`](#tidb_analyze_version-span-classversion-mark-v510-span)
+    - [`tidb_enable_rate_limit_action`](#tidb_enable_rate_limit_action)
+
+To avoid accumulating too many status files for alarms, currently, TiDB only retains the status files generated during the recent five alarms by default. You can adjust this number by configuring the system variable [`tidb_memory_usage_alarm_keep_record_num`](/system-variables.md#tidb_memory_usage_alarm_keep_record_num-new-in-v640).
```

Comment:
```suggestion
To avoid accumulating too many status files for alarms, TiDB only retains the status files generated during the recent five alarms by default. You can adjust this number by configuring the system variable [`tidb_memory_usage_alarm_keep_record_num`](/system-variables.md#tidb_memory_usage_alarm_keep_record_num-new-in-v640).
```